### ISSUE TITLE
Wrong error message during finetuning

### ISSUE THREAD

**Issue by dakshvar22** [RasaHQ/rasa#10246]
### Rasa Open Source version

3.0.0rc2

### Python version

3.8

### What operating system are you using?

OSX

### What happened?

Started with a `rasa init` project which trained an initial model. I made changes to the NLU training data by adding a new intent:

```
- intent: new_intent
  examples: |
    - yes i am new
    - can i break you
    - i don't think i want to do this
    - but its exciting
```

Ran `rasa train --finetune`. The process exits with:

```
InvalidConfigException: Cannot finetune because more than just the 'epoch' keys have been changed in the configuration. Please revert your configuration and only change the 'epoch' settings where needed.
```

whereas according to 2.8.x, the log should have been:

```
2021-11-16 15:44:23 INFO     rasa.model  - Data (nlu_labels) for finetune section changed.
NLU model can not be finetuned.
```

which is the correct info as we only changed the data and added a new intent label which should disallow finetuning.

### Command / Request

```shell
rasa train --finetune
```

**Comment by dakshvar22** [comment]
It is correct that if new intents, entities, entity groups, entity roles, and action names have been added compared to the initial training phase, finetuning is not possible and it should return an appropriate message that new labels have been added (you can try `2.8.x` branch to test this).

This does not apply to adding a new training example to **existing** intents, entities, etc. 

To answer your questions:

> Are changes to the nlu.yml counted as "configuration" changes?

No, they are not configuration changes. Configuration changes are only the ones that are made to `config.yml` file.

> did you make any configuration changes?

Nope 

> then perhaps we should re-arrange the order in which we check the cases listed above?

The checks are fine but I suspect that an addition of a new training example to existing intent is being treated as an addition of a new intent which is wrong. 😅

**Comment by carlad** [comment]
Thanks @dakshvar22 😄 

I did have a question:

> The checks are fine but I suspect that an addition of a new training example to existing intent is being treated as an addition of a new intent which is wrong

If I understand your original comment, what you did was add an entirely new intent:
```
- intent: new_intent
  examples: |
    - yes i am new
    - can i break you
    - i don't think i want to do this
    - but its exciting
 ```

And that this should trigger the 3rd check (the nlu check), and show the message:
```
"Cannot finetune because NLU training data contains new labels "
"or does not contain any examples for some known labels. " ...
```

Did you add anything to an existing intent?

In any case, looking at the [code](https://github.com/RasaHQ/rasa/blob/main/rasa/graph_components/validators/finetuning_validator.py#L137), the 1st check (the config check), which raises the message you saw, only looks at the `config` values of the fingerprint. I will see if I can replicate the your error using the steps you describe.

**Comment by dakshvar22** [comment]
Terribly sorry @carlad ! This is my mistake. During the QA, I did add a new intent with examples to it and realized that finetuning wouldn't be possible in that case.. So, just added a new example to an existing intent but forgot to remove the data for new intent that I previously added and that's why I kept getting the error.. i just tried adding a new example to the existing intents of a project and tried finetuning, it works!

Again sorry for the confusion caused here 😞

**Comment by dakshvar22** [comment]
What I can still indeed reproduce is this:

1. Create a new project with `rasa init`
2. Add a new intent like:
```
- intent: new_intent
  examples: |
    - yes i am new
    - can i break you
    - i don't think i want to do this
    - but its exciting
```
3. Try finetuning with `rasa train --finetune`

I get the error saying `InvalidConfigException: Cannot finetune because more than just the 'epoch' keys have been changed in the configuration. Please revert your configuration and only change the 'epoch' settings where needed.`

Whereas, I would expect - `InvalidConfigException: Cannot finetune because NLU training data contains new labels or does not contain any examples for some known labels. Please make sure that the NLU data that you use for finetuning contains at least one example for every label (i.e. intent, action name, ...) that was included in the NLU data used for training the model which we attempt to finetune now. Moreover, you must not add labels that were not included during training before.`

This should be fixed in this issue IMO.

Does that help clarify?

**Comment by carlad** [comment]
No problem! It's been a great opportunity for me to understand finetuning validation and fingerprints a little more.

It still doesn't explain why you got the config mismatch error message, rather than the nlu mismatch error message, if nothing in your `config.yml` changed... 🤔

**Comment by dakshvar22** [comment]
> It still doesn't explain why you got the config mismatch error message, rather than the nlu mismatch error message

Yep, that should be investigated and fixed 😄

**Comment by dakshvar22** [comment]
Okay, I have more evidence for you... This only happens when you use the config that comes with autoconfiguration of `rasa init` as it is. If you uncomment the lines of the config but still use the same pipeline of the config in the above setup, you'll get the correct error message. Definitely something to do with how config is being fingerprinted across multiple runs?

**Comment by ka-bu** [comment]
🤔  I'm suspecting that the importer will add the new intent to the domain for us and that afterwards (at that point, given those updates) we have no way to distinguish whether the user changed only the nlu data and/or the domain -- There's definitely some merging going on with retrieval intents [here](https://github.com/RasaHQ/rasa/blob/2e82c9caab989348d57e74d918cc4286f85778a5/rasa/shared/importers/importer.py#L303)

**Comment by ka-bu** [comment]
Problem seems to be that in the default project, we fill in ["debug_plots", "augmentation_factor"] but for some reason those parameters are not set when running rasa init -- Guess that comes from some mismatch of defaults for those cli parameters.
